artemis/fallback_api_cache.py

```diff
@@ -0,0 +1,93 @@
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Any, Callable, Dict, List
+
+from requests import Response
+from requests_cache import CachedSession
+from requests_cache.backends.filesystem import FileCache
+
+from artemis import utils
+from artemis.config import Config
+
+
+class InvalidResponseException(Exception):
+    pass
+
+
+@dataclass
+class CachedURL:
+    url: str
+    validator: Callable[[Dict[Any, Any]], bool]
+
+    def get(self) -> Response:
+        return FallbackAPICache.get(self.url)
+
+
+class FallbackAPICache:
+    LOGGER = utils.build_logger(__name__)
+
+    CACHE = CachedSession(
+        backend=FileCache("/cache"),
+        cache_control=False,
+        expire_after=timedelta(seconds=3600),
+        allowable_codes=[200],
+        stale_if_error=True,
+    )
+
+    URL_JOOMLA_LATEST_RELEASE = CachedURL(
+        "https://api.github.com/repos/joomla/joomla-cms/releases/latest",
+        lambda item: "url" in item and "author" in item,
+    )
+
+    URL_WORDPRESS_TAGS = CachedURL(
+        "https://api.github.com/repos/WordPress/WordPress/git/refs/tags",
+        lambda item: len(item) > 10 and "ref" in item[0] and "url" in item[0],
+    )
+
+    URL_WORDPRESS_STABLE_CHECK = CachedURL(
+        "https://api.wordpress.org/core/stable-check/1.0/", lambda item: item["1.0.2"] == "insecure"
+    )
+
+    URL_WORDPRESS_PLUGINS_LIST = CachedURL(
+        "https://api.wordpress.org/plugins/info/1.2/?action=query_plugins&request[page]=1&request[per_page]=1000",
+        lambda item: "info" in item and "page" in item["info"] and "plugins" in item and "name" in item["plugins"][0],
+    )
+
+    @classmethod
+    def get(cls, url: str, allow_unknown: bool = False) -> Response:
+        headers = {}
+        if Config.Miscellaneous.CUSTOM_USER_AGENT:
+            headers["User-Agent"] = Config.Miscellaneous.CUSTOM_USER_AGENT
+
+        response = FallbackAPICache.CACHE.get(url, headers=headers)
+
+        found = False
+        for item in cls.urls():
```

Comment:
If we care about what URLs to be accessed with this class, check should be done before sending GET request